rxnetty/src/main/java/io/reactivex/netty/protocol/http/UnicastContentSubject.java

```diff
@@ -138,7 +137,14 @@ private UnicastContentSubject(final State<T> state, long noSubscriptionTimeout,
      */
     public boolean disposeIfNotSubscribed() {
         if (state.casState(State.STATES.UNSUBSCRIBED, State.STATES.DISPOSED)) {
-            state.bufferedSubject.subscribe(Subscribers.empty()); // Drain all items so that ByteBuf gets released.
+            state.bufferedSubject.subscribe(new Action1<T>() {
+                @Override
+                public void call(T t) {
+                    if (t instanceof ReferenceCounted) {
```

Comment:
> Is there a specific reason it gets retained in the subject?

Good question!
The reason is issue #264 `ByteBuf` is released after calling `onNext` on this subject. So, in case we buffer (no eager subscription to this subject), after `onNext` on this subject returns, the `ByteBuf` will have refCnt = 0 if we don't retain first.

There are two distinct scenarios:

- Eager subscription; we don't buffer and the subject is a pass through. The subject retain once, release once. The code invoking `onNext` on this Subject releases when `onNext` returns.
- No eager subscription; we buffer after doing a `retain`. If we don't retain, `ByteBuf` after `onNext` returns, ref count will be 0. After buffering, whenever this subject is subscribed to (dispose does an empty subscriber), the `ByteBuf` should be released so that user code is agnostic to this buffering (w/o buffering, user does not explicitly release)

After Issue #264 is fixed (its a behavior change), we will be removing the retain but still do release in dispose, just once.